### ISSUE TITLE
Suspicious assignments: further enhancements 

### ISSUE THREAD

**Issue by kalessil** [kalessil/phpinspectionsea#220]
```php
function immediateOverrides()
{
    /* Case 1: possible override */
    if ($x) {
        $y = '';
    }
    $y = ''; // <- reported, the value is not depending onto the target variable; perhaps else was intended to use

    /* false-positive: 2nd write is optional */
    $t = '';
    if ($x) {
        $t = '';
    }

    /* Case 2: guaranteed override */
    $z = '';
    $z = $y; // <- reported, the value is not depending onto the target variable

    return [$y, $z, $t];
}
```
